src/ApiPort.VisualStudio/DteProjectExtensions.cs

```diff
@@ -1,74 +1,189 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using ApiPortVS.Resources;
 using EnvDTE;
-using Microsoft.Fx.Portability;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace ApiPortVS
 {
     internal static class DteProjectExtensions
     {
         private const string FSharpProjectKindString = "{f2a71f9b-5d33-465a-a702-920d77279786}";
 
-        public static string GetBuildOutputFileName(this Project project)
+        /// <summary>
+        /// Tries to fetch output items if it uses Common Project System then
+        /// tries to fetch output items by retrieving FinalBuildOutput
+        /// location using code snippet from:
+        /// https://github.com/Microsoft/visualfsharp/blob/master/vsintegration/tests/unittests/Tests.ProjectSystem.Miscellaneous.fs#L168-L182
+        /// </summary>
+        /// <returns>null if it is unable to retrieve VS configuration objects</returns>
+        public static async Task<IEnumerable<string>> GetBuildOutputFilesAsync(this Project project, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var fileName = project.Properties.Item("OutputFileName").Value.ToString();
+            var output = await project?.GetBuildOutputFilesFromCPSAsync(cancellationToken);
 
-            return Path.Combine(project.GetBuildOutputPath(), fileName);
+            if (output != null)
+            {
+                return output;
+            }
+
+            var configuration = GetVsProjectConfiguration(project);
+
+            if (configuration == null)
+            {
+                return null;
+            }
+            if (!(configuration is IVsProjectCfg2))
```

Comment:
This could be rewritten with pattern matching; ie `if(configuration is IVsProjectCfg2 configuration2)`. You'll have to swap the branching here, though